### PR TITLE
Allow users to include additional Markdown pages

### LINKED ISSUE #316: Allow users to include additional markdown pages
YARD, a similiar tool for Ruby, allows the user to include additional pages into the generated site, such as this [getting started](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md) page on their site.

I like this feature as it allows me to include documents that detail more general architectural information in my documentation, and I think it would make a strong addition to ExDoc. :)

Thanks,
Louis

### PR DESCRIPTION
**Pull request by milmazz**
This PR allows the user to include additional Markdown pages, via command line you can use as many `--extra` (or the alias `-e`) as you want:

```bash
$ bin/elixir ../ex_doc/bin/ex_doc "EEx" "1.1.0-beta" "lib/eex/ebin" -m "EEx" -u "https://github.com/elixir-lang/elixir" --source-ref "64bb812f15e8b9ec078f76cfd8e26578bf2ddaa0"  -o doc/eex -p http://elixir-lang.org/docs.html \
--readme README.md -e CHANGELOG.md -e CONTRIBUTING.md

$ ls doc/eex/{README,CHANGELOG,CONTRIBUTING}.html         
doc/eex/CHANGELOG.html doc/eex/CONTRIBUTING.html doc/eex/README.html
```

For `Mix` there is a new options called `extras: `, example:

```
# file: mix.exs
docs: [source_ref: "v#{@version}", main: "Ecto",
            source_url: "https://github.com/elixir-lang/ecto",
            readme: "README.md", extras: ["CHANGELOG.md"]]]
```

Closes #316 

/cc @josevalim @Dignifiedquire

### CODE REVIEW

**Comment by Dignifiedquire** [comment]
Love it, even styling wise I think this is good for now!

File: lib/ex_doc/cli.ex [hunk 1]
```diff
@@ -25,6 +34,30 @@ defmodule ExDoc.CLI do
     end
   end
 
+  # TODO: At some point we need to mark the --readme option
```
Comment (josevalim) [inline lib/ex_doc/cli.ex]: We can deprecate it now. We are before 1.0. :)

Comment (josevalim) [inline lib/ex_doc/cli.ex]: Actually, just remove it.

File: lib/mix/tasks/docs.ex [hunk 2]
```diff
@@ -101,4 +106,17 @@ defmodule Mix.Tasks.Docs do
       true -> docs
     end
   end
+
+  # At some point we need to mark :readme option as
```
Comment (josevalim) [inline lib/mix/tasks/docs.ex]: Same here. Just remove README, people can update their mix docs. Benefits of being before 1.0. :)

**Comment by josevalim** [comment]
AWesome work @milmazz! I have added two comments, feel free to remove the README option, and let's ship it!

**Comment by lpil** [comment]
Yay!

**Pull request merged**